apps/admin-x-activitypub/src/views/Feed/components/FeedList.tsx

```diff
@@ -34,6 +34,49 @@ const FeedList:React.FC<FeedListProps> = ({
     const loadMoreRef = useRef<HTMLDivElement | null>(null);
     const endLoadMoreRef = useRef<HTMLDivElement | null>(null);
 
+    // Group activities by their original post content
+    type GroupedActivity = { mainActivity: Activity; reposters: { actor: Activity['actor']; activity: Activity }[] };
+    const groupedActivities = activities.reduce((acc, activity) => {
+        // Use the post content URL as the grouping key since all reposts share the same base post
+        const groupKey = activity.object.url || activity.id;
+
+        if (!acc[groupKey]) {
+            acc[groupKey] = {
+                mainActivity: activity,
+                reposters: []
+            };
+
+            if (activity.type === 'Announce' && activity.actor) {
+                acc[groupKey].reposters.push({
+                    actor: activity.actor,
+                    activity: activity
+                });
+            }
+        } else {
+            if (activity.type === 'Announce' && activity.actor) {
+                acc[groupKey].reposters.push({
+                    actor: activity.actor,
+                    activity: activity
+                });
+            } else if (activity.type === 'Create') {
+                acc[groupKey].mainActivity = activity;
+            }
+        }
+
+        return acc;
+    }, {} as Record<string, GroupedActivity>);
+
+    const consolidatedActivities = (Object.values(groupedActivities) as GroupedActivity[])
+        .map((group: GroupedActivity) => ({
+            mainActivity: group.mainActivity,
+            reposts: group.reposters.map(r => r.activity),
+            mostRecentDate: Math.max(
+                new Date(group.mainActivity.object.published || 0).getTime(),
+                ...group.reposters.map(r => new Date(r.activity.object.published || 0).getTime())
+            )
+        }))
+        .sort((a, b) => b.mostRecentDate - a.mostRecentDate);
```

Comment:
This sorting is redundant as `r.activity.object.published` is same as `group.mainActivity.object.published`.  `r.activity.object.published` is not the date/time at which the post was reposted but the date at which the original post was created. 
I think the feed will be sorted correctly even if we remove this sorting because we receive the notes in order(recent first) from backend.